src/web-auth/captcha.js

```diff
@@ -81,29 +93,57 @@ function scriptForRecaptchaProvider(provider, lang, callback) {
         '&onload=' +
         callback
       );
+    case HCAPTCHA_PROVIDER:
+      return (
+        'https://js.hcaptcha.com/1/api.js?hl=' +
+        lang +
+        '&onload=' +
+        callback
+      );
+    case FRIENDLY_CAPTCHA_PROVIDER:
+      return 'https://cdn.jsdelivr.net/npm/friendly-challenge@0.9.12/widget.min.js';
     /* istanbul ignore next */
     default:
       throw new Error('Unknown captcha provider');
   }
 }
 
-function injectRecaptchaScript(element, opts, callback) {
-  var callbackName = 'recaptchaCallback_' + Math.floor(Math.random() * 1000001);
+function injectCaptchaScript(element, opts, callback) {
+  var providerName;
+  switch (opts.provider) {
+    case RECAPTCHA_ENTERPRISE_PROVIDER:
+      providerName = 'recaptcha';
+      break;
+    case RECAPTCHA_V2_PROVIDER:
+      providerName = 'recaptcha';
+      break;
+    case HCAPTCHA_PROVIDER:
+      providerName = 'hcaptcha';
+      break;
+    case FRIENDLY_CAPTCHA_PROVIDER:
+      providerName = 'friendly_captcha';
+      break;
+  }
+  var callbackName = providerName + 'Callback_' + Math.floor(Math.random() * 1000001);
   window[callbackName] = function() {
     delete window[callbackName];
     callback();
   };
   var script = window.document.createElement('script');
-  script.src = scriptForRecaptchaProvider(
+  script.src = scriptForCaptchaProvider(
     opts.provider,
     opts.lang,
     callbackName
   );
   script.async = true;
+  script.defer = true;
```

Comment:
Oddly enough, i double checked the Recaptcha docs and their example also uses `defer`. 
https://developers.google.com/recaptcha/docs/display
```
<script src="https://www.google.com/recaptcha/api.js?onload=onloadCallback&render=explicit"
    async defer>
</script>
```

Reply:
I am a bit worried this could have unexpected effects, for what I understand `defer` moves the loading to when after the page is done loaded, while `async` does it instantly. ,but asynchronously and non-blocking.

By adding defer, we are delaying the loading of the script which could, potentially, change the experience in the consuming applications.

Do we need defer for the captcha's being added here to work?

Reply:
Just for reference:

> The defer attribute may be specified even if the async attribute is specified, to cause legacy Web browsers that only support defer (and not async) to fall back to the defer behavior instead of the synchronous blocking behavior that is the default.

https://www.w3.org/TR/2011/WD-html5-20110525/scripting-1.html#attr-script-async

That being said, I'm totally fine to preserve Recaptcha functionality being and not make unnecessary changes.